### PR TITLE
IGNITE-13507 NullPointerException on tx recovery

### PR DESCRIPTION
**Pull request by pvinokurov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java [hunk 1]
```diff
@@ -3353,9 +3353,14 @@ else if (tx.setRollbackOnly())
...
-            return node.isClient() && mvccCrd != null && mvccCrd.nodeId() != null &&
+            ClusterNode mvccCrdNode = null;
+
+            if (mvccCrd != null && mvccCrd.nodeId() != null)
+                mvccCrdNode = cctx.node(mvccCrd.nodeId());
+
+            return node.isClient() && mvccCrdNode != null &&
                 (cctx.kernalContext().coordinators().mvccEnabled() ||
-                    !IgniteFeatures.nodeSupports(cctx.node(mvccCrd.nodeId()), IgniteFeatures.MVCC_TX_RECOVERY_PROTOCOL_V2));
+                    !IgniteFeatures.nodeSupports(mvccCrdNode, IgniteFeatures.MVCC_TX_RECOVERY_PROTOCOL_V2));
```
Comment (alamar) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java:3363]: Please remind me why do we need this at all? Apache Ignite does not have compatibility between versions.

Can we get rid of feature check entirely, assume it as "always true"?

Comment (pvinokurov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java:3363]: I am not sure that it is related to comatibility. It was introduced as an optimization within fixes for https://issues.apache.org/jira/browse/IGNITE-12350

Comment (alamar) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java:3363]: Please try to remove IgniteFeatures use, see if nothing is broken.

Comment (pvinokurov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java]: Using IgniteFeatures was removed. It looks like nothing is broken

Comment (pvinokurov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java]: Also, I changed the name of the test method

**Pull request closed without merging**